sympy/polys/tests/test_polytools.py

```diff
@@ -1184,9 +1184,36 @@ def test_Poly_as_expr():
 
 
 def test_Poly_lift():
-    assert Poly(x**4 - I*x + 17*I, x, gaussian=True).lift() == \
-        Poly(x**16 + 2*x**10 + 578*x**8 + x**4 - 578*x**2 + 83521,
-             x, domain='QQ')
+    p = Poly(x**4 - I*x + 17*I, x, gaussian=True)
+    assert p.lift() == Poly(x**8 + x**2 - 34*x + 289, x, domain='QQ')
```

Comment:
For anyone reviewing this the basic idea is that if `p` is a polynomial over an extension field over `K` and `P = p.lift()` then `P` is a polynomial with coefficients in `K` such that `p` divides `P`. Therefore the roots of `P` are a superset of the roots of `p`:
```python
In [10]: p = Poly(x**4 - I*x + 17*I, x)

In [11]: p
Out[11]: Poly(x**4 - I*x + 17*I, x, domain='ZZ_I')

In [12]: P = p.lift()

In [13]: P.rem(p)
Out[13]: Poly(0, x, domain='ZZ_I')

In [17]: for r in p.nroots(2): print(r)
-1.9 + 0.82*I
-0.74 - 1.9*I
0.82 + 1.8*I
1.8 - 0.73*I

In [18]: for r in P.nroots(2): print(r)
-1.9 - 0.82*I
-1.9 + 0.82*I
-0.74 - 1.9*I
-0.74 + 1.9*I
0.82 - 1.8*I
0.82 + 1.8*I
1.8 - 0.73*I
1.8 + 0.73*I
```